tests/Nox.Integration.Tests/DatabaseIntegrationTests/NoxCommonTestCaseFactory.cs

```diff
@@ -1019,4 +1019,45 @@ public void LocalizedEntitiesBeingGenerated()
         Assert.Equal(testEntity.TextFieldToLocalize.Value, text);
         Assert.Equal(testEntity.CultureCode.Value, culture);
     }
+    
+    public void AutoNumberedEntitiesBeingGenerated()
+    {
+
+        var idValue = 10;
+        var propertyValue = 20;
+        
+        var text = "TX";
+
+        var stru = DataContext.Database.GetConnectionString(); //("SELECT * FROM information_schema.sequences;");
+
+        var newItem = new TestEntityForAutoNumberUsages()
+        {
+            TextField = Text.From(text),
+        };
+
+        DataContext.TestEntityForAutoNumberUsages.Add(newItem);
+        
+        var newItem2 = new TestEntityForAutoNumberUsages()
+        {
+            TextField = Text.From(text),
+        };
+
+        DataContext.TestEntityForAutoNumberUsages.Add(newItem2);
+        DataContext.SaveChanges();
+
+        // Force the recreation of DataContext and ensure we have fresh data from database
+        _dbContextFixture.RefreshDbContext();
+
+        var testEntity = DataContext.TestEntityForAutoNumberUsages.First();
+        var testEntity2 = DataContext.TestEntityForAutoNumberUsages.OrderBy(e=>e.Id).Last();
+
+        
+        
+        
+        Assert.Equal(idValue, testEntity.Id.Value);
+        Assert.Equal(idValue + 2, testEntity2.Id.Value);
```

Comment:
```suggestion
        Assert.Equal(testEntity2.Id.Value, idValue + 2);
```

Reply:
As far as I check the Equal method implementation left should be expected, right should be actual. What do you think ? 
`		/// <summary>
		/// Verifies that two objects are equal, using a default comparer.
		/// </summary>
		/// <typeparam name="T">The type of the objects to be compared</typeparam>
		/// <param name="expected">The expected value</param>
		/// <param name="actual">The value to be compared against</param>
		public static void Equal<T>(
#if XUNIT_NULLABLE
			[AllowNull] T expected,
			[AllowNull] T actual) =>
#else
			T expected,
			T actual) =>
#endif
				Equal(expected, actual, GetEqualityComparer<T>());`

Reply:
You are right, I was seeing the other Asserts in this file, but you can leave it like this, thanks.